test/service-object.ts

```diff
@@ -114,13 +115,26 @@ describe('ServiceObject', () => {
       // tslint:disable-next-line:variable-name
       const FakePromise = () => {};
       const config = extend({}, CONFIG, {
-        parent: {
-          Promise: FakePromise,
-        },
+        parent: {Promise: FakePromise, requestModule: {}},
       });
       const serviceObject = new ServiceObject(config) as FakeServiceObject;
       assert.strictEqual(serviceObject.Promise, FakePromise);
     });
+
+    it('should inherit the parents requestModule', () => {
+      const serviceObject = new ServiceObject(CONFIG);
+      assert.strictEqual(
+          serviceObject.requestModule, CONFIG.parent.requestModule);
+    });
+
+    it('should throw if a requestModule is not passed to the constructor',
```

Comment:
Thank goodness for auto-formatters, 'cause imagine trying to explain the rules of the next several lines to a contributor 🤣

Reply:
software is terrible.